src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/GhostExprTyping.scala

```diff
@@ -141,12 +141,20 @@ trait GhostExprTyping extends BaseTyping { this: TypeInfoImpl =>
     case _: PAccess | _: PPredicateAccess => AssertionT
 
     case expr : PGhostCollectionExp => expr match {
-      case PCardinality(_) => IntT
-      case PMultiplicity(_, _) => IntT
+      // The result of integer ghost expressions is unbounded (UntypedConst)
+      case PCardinality(_) => IntT(UntypedConst)
+      case PMultiplicity(_, _) => IntT(UntypedConst)
       case PIn(_, _) => BooleanT
       case expr : PSequenceExp => expr match {
-        case PRangeSequence(_, _) => SequenceT(IntT)
-        case PSequenceAppend(left, _) => exprType(left)
+        case PRangeSequence(_, _) => SequenceT(IntT(UntypedConst))
+
+        case PSequenceAppend(left, right) =>
+          val lType = exprType(left)
+          val rType = exprType(right)
+          typeMerge(lType, rType) match {
+            case Some(seq@SequenceT(_)) => seq
+            case _ => lType
```

Comment:
In this case, you should maybe throw a violation.

Reply:
The reason why I did not throw a violation is that Gobra would terminate with an exception whereas if I let it proceed, the well-defidedness checker would detect the error and send a message and, this way, it would provide useful information to the user instead of seeing a stack trace. Would it be better if I returned `UnknownType` instead of   `lType`?

Reply:
Usually, the pattern is that the well-def functions check if a node can be typed and the typing function then returns the type. In partcular, the typing function is only evaluated if the well-def function returned that the function is safe to type. Following that pattern, the typing functions should throw an exception if something problematic occurs, since the well-def function should have checked and handled this,

Reply:
Solved in 76be76a